### PR TITLE
Feat:Contribution weekly section

### PR DESCRIPTION
**Pull request by cp-priyanka-g**
## Review checklist

**Note:** Make sure all points should be checked before submiting PR for review, otherwise PR will be cosidered invalid.

- [x] Follow guidelines for creating branch name, adding commit messages or PR title
- [x] Add MR description if applicable
- [x] Follow guidelines for naming conventions everywhere (i.e files/folders, data structures)
- [x] Reuse code (if the same code is written twice, make it common and reuse it at both places)
- [x] Remove unused or commented code if not required
- [x] Make MR self-approved - review MR as a reviewer and give it self-approval if everything is ok, if not make the required changes
- [ ] In progress MR should be marked as draft
- [x] Make MR mark as ready before submitting it for review
- [x] Add images, videos, or gifs of UI 
- [x] Add page speed score screenshot of UI (It should be > 85 for mobile and > 90 for desktops)

**Design should be tested on:**
- [x] Inspector (all iPhones and android devices like samsung, motorola)
- [x] Desktop (chrome, firefox, safari)
- [ ] Laptop / Macbook  (chrome, firefox, safari)
- [ ] iPhone(safari)
- [x] Available android devices  (chrome, firefox)
- [x] iPad(safari)
- [ ] Tablet (chrome, firefox)

**Description:**
Create Weekly update section.
Video:
[weeklyUpdate.webm](https://github.com/canopas/canopas-website/assets/98794636/f191f730-13b3-4c21-99fa-b219cadd795e)

**Desktop:**
image

**Mobile:**
image

**Mobile Performance:**
image

**Desktop Performance:**
image

### CODE REVIEW

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        },
+      ],
+    };
+  },
+  components: {
+    Swiper,
+    SwiperSlide,
+  },
+  mounted() {
+    this.height = window.innerHeight;
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/WeeklyUpdate.vue:225]: Is this required?

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    >
+      <div>
+        <span
+          class="tw-text-[1.875rem] lg:tw-text-[3.4375rem] tw-leading-[2.4375rem] lg:tw-leading-[5.15625rem] tw-text-black-core/[.87] tw-font-inter-bold"
+        >
+          Our Weekly Tech Updates</span
+        >
+      </div>
+      <div
+        class="tw-mb-8 tw-mt-2.5 lg:tw-mt-6 tw-text-center tw-text-[1rem] lg:tw-text-[1.5rem] tw-leading-[1.5rem] lg:tw-leading-[2.25rem] tw-text-black-core/[0.87] lg:tw-text-black-core/[0.6] tw-font-inter-regular lg:tw-font-inter-medium"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/WeeklyUpdate.vue:14]: Is Font color different from lg devices?

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        :slidesPerView="1.1"
+        :spaceBetween="0"
+        :effect="'cards'"
+        :grabCursor="true"
+        @swiper="setSwiperRef"
+        @slideChange="activeIndex = swiperRef.activeIndex"
+        class="swiper-container"
+      >
+        <swiper-slide v-for="(weekly, index) in weeklies" :key="index">
+          <a class="tw-w-[380px] tw-h-[460px]" :href="weekly.url">
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/WeeklyUpdate.vue:51]: Couldn't card render without defining manual height and width?

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    <!-- Mobile UI end -->
+    <!-- Desktop UI start -->
+    <div
+      class="tw-hidden md:tw-flex tw-justify-around tw-m-auto tw-mb-[7rem] tw-mt-8 md:tw-my-[3.75rem] tw-w-[100%] md:tw-w-[95%] xl:tw-w-[65%] 2xl:tw-w-[55%] xll:tw-w-[30%] tw-rounded-[30px] tw-py-[5px]"
+    >
+      <a
+        v-for="(weekly, index) in weeklies"
+        :key="index"
+        target="_blank"
+        class="tw-hidden md:tw-block tw-flex-[16.66%] tw-gap-x-4 tw-m-0 tw-w-full tw-rounded-[25px] tw-px-[5px] tw-py-[0.7rem] md:tw-py-[15px] tw-text-center tw-tracking-[1px] tw-text-[0.75rem] sm:tw-text-[1rem] md:tw-text-[1.25rem] tw-leading-[0.938rem] md:tw-leading-[1.1875rem] tw-font-inter-regular portfolio-nav md:hover:tw-bg-gradient-[270.11deg] md:hover:tw-from-[#ff835b] md:hover:tw-to-[#f2709c] md:hover:tw-text-white"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/WeeklyUpdate.vue:92]: `portfolio-nav` class used?

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue (hunk 2 above)
Comment (cp-priyanka-g) [reply vue-frontend/src/components/contributions/WeeklyUpdate.vue:14]: yes , different opacity in mobile and dektop

File: vue-frontend/src/components/contributions/WeeklyUpdate.vue [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+            : 'tw-border-[1px] tw-border-solid tw-border-transparent tw-text-black-core/[0.6]'
+        "
+        @click="scrollToCard(index)"
+      >
+        {{ weekly.category }}
+      </a>
+    </div>
+
+    <ul
+      @scroll="handleScroll"
```
Comment (cp-sumi-k) [inline vue-frontend/src/components/contributions/WeeklyUpdate.vue:108]: Remove it from here and add listener to mounted hook

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**